llvm/lib/CodeGen/InterleavedAccessPass.cpp

```diff
@@ -249,11 +249,48 @@ static bool isReInterleaveMask(ShuffleVectorInst *SVI, unsigned &Factor,
   return false;
 }
 
+/// Return true if it's a non-all-zeros, interleaving mask. For instance,
+/// 111000111000 is interleaved from three 1010 masks.
+/// \p SubMask returns the mask of individual lane.
+static bool isInterleavedConstantMask(unsigned Factor, ConstantVector *Mask,
+                                      SmallVectorImpl<Constant *> &LaneMask) {
+  unsigned LaneMaskLen = LaneMask.size();
+  if (auto *Splat = Mask->getSplatValue()) {
+    // All-zeros mask.
+    if (Splat->isZeroValue())
+      return false;
```

Comment:
Why do we reject an all 0s mask? Hopefully its gets removed by InstCombine earlier, but if not it should still be valid to make a segment load/store with a 0 mask and it would avoid the interleave/deinterleave lowering in SelectionDAG.

Reply:
I thought it'll got removed anyway by one of the IR Passes in the codegen pipeline. But with a quick check that doesn't seem to be the case. I'll let this case through in my next update.

Reply:
It's done.